src/DynamoRevit/DynamoRevit.cs

```diff
@@ -477,14 +477,42 @@ private static RevitDynamoModel InitializeCoreModel(DynamoRevitCommandData comma
         internal static Version PreloadAsmFromRevit()
         {
             var asmLocation = AppDomain.CurrentDomain.BaseDirectory;
-            Version libGversion = findRevitASMVersion(asmLocation);
+            Version libGVersion = findRevitASMVersion(asmLocation);
             var dynCorePath = DynamoRevitApp.DynamoCorePath;
             // Get the corresponding libG preloader location for the target ASM loading version.
             // If there is exact match preloader version to the target ASM version, use it, 
             // otherwise use the closest below.
-            var preloaderLocation = DynamoShapeManager.Utilities.GetLibGPreloaderLocation(libGversion, dynCorePath);
+            var preloaderLocation = DynamoShapeManager.Utilities.GetLibGPreloaderLocation(libGVersion, dynCorePath);
+
+            // [Tech Debt] (Will refactor the code later)
+            // The LibG version maybe different in Dynamo and Revit, using the one which is in Dynamo.
+            Version preLoadLibGVersion = PreloadLibGVersion(preloaderLocation);
             DynamoShapeManager.Utilities.PreloadAsmFromPath(preloaderLocation, asmLocation);
-            return libGversion;
+            return preLoadLibGVersion;
+        }
+
+        // [Tech Debt] (Will refactor the code later)
+        /// <summary>
+        /// Return the preload version of LibG.
+        /// </summary>
+        /// <param name="preloaderLocation"></param>
+        /// <returns></returns>
+        internal static Version PreloadLibGVersion(string preloaderLocation)
+        {
+            preloaderLocation = preloaderLocation.Substring(preloaderLocation.LastIndexOf('\\') + 1);
```

Comment:
what about https://stackoverflow.com/questions/27019082/how-can-i-get-the-last-folder-from-a-path-string to get the directory name?